include/mbedtls/x509_crt.h

```diff
@@ -501,6 +501,70 @@ int mbedtls_x509_crt_verify_restartable( mbedtls_x509_crt *crt,
                      void *p_vrfy,
                      mbedtls_x509_crt_restart_ctx *rs_ctx );
 
+#if defined(MBEDTLS_X509_TRUSTED_CERTIFICATE_CALLBACK)
+/**
+ * \brief               The type of trusted certificate callbacks.
+ *
+ *                      Callbacks of this type are passed to and used by the CRT
+ *                      verification routine mbedtls_x509_crt_verify_with_cb()
+ *                      when looking for trusted signers of a given certificate.
+ *
+ *                      On success, the callback returns a list of trusted
+ *                      certificates to be considered as potential signers
+ *                      for the input certificate.
+ *
+ * \param p_ctx         An opaque context passed to the callback.
+ * \param child         The certificate for which to search a potential signer.
+ *                      This must point to a readable certificate.
```

Comment:
Changed

Reply:
@mpg @jarlamsa I think there is a misunderstanding here: The `child` pointer must point to a readable CRT, and the `candidate_cas` ptr must not be `\c NULL`. This is to be distinguished from the value of `*candidate_cas` on success - but even the latter may be `NULL` if the functions returns successfully but didn't find any candidates.

Reply:
Those are "must" from the APIs point of view and "will" from the users point of view. Both are in sense valid, but which are we using? In the following note the must is referring that what the callback must do.

Reply:
@hanno-arm I think we're talking about different things here, I'm really not talking about returned values or the distinction between `canditates_cas` and `*candidate_cas`, but rather about the difference between documenting a normal API function and the type of a callback.

For API functions, we (Mbed TLS) are providing a function to users and telling them what they must do when using this function. For callbacks, users are providing a function to us, are we're telling them what we promise we'll do when calling that function, so that they now what assumptions they can rely on when writing the function.

For me, the point is who's talking to whom. Since we're writing the documentation, it feels weird to me to write "must" for documenting requirements on ourselves. We should be talking to the user, not to ourselves.

Reply:
@mpg @jarlamsa Yes, that makes sense - thank you for elaborating, and sorry for missing that point in the first place!